src/components/DropDownButton.tsx

```diff
@@ -133,6 +135,24 @@ class DropDownButton extends React.Component<
 		};
 	}
 
+	componentDidMount() {
+		window.document.addEventListener('click', this.handleOutsideClick);
```

Comment:
NIT: How about registering a single listener for all dropdowns or attach the listener only when the dropdown opens?

Reply:
@thgreasi If it is a single listener it will complicate things a bit and break isolation IMO. Regarding attaching the event listener when it opens, I was also thinking if it's better like that or not, but I sort of leaned towards registering listeners once. I am reconsidering it now, as it might be a bit "cleaner".